aries_vcx/src/utils/devsetup.rs

```diff
@@ -223,38 +198,36 @@ impl Drop for SetupPoolConfig {
 
 impl SetupPoolMocks {
     pub async fn init() -> SetupPoolMocks {
-        setup();
-        setup_indy_env(false).await;
+        init_test_logging();
+        reset_global_state();
+
+        setup_indy_env().await;
         enable_pool_mocks();
         SetupPoolMocks {}
     }
 }
 
-impl Drop for SetupPoolMocks {
-    fn drop(&mut self) {
-        tear_down()
-    }
-}
-
 impl SetupIndyMocks {
     pub fn init() -> SetupIndyMocks {
-        setup();
-        settings::set_config_value(settings::CONFIG_ENABLE_TEST_MODE, "true");
+        init_test_logging();
+        reset_global_state();
+
+        enable_indy_mocks();
         settings::get_agency_client_mut().unwrap().enable_test_mode();
         SetupIndyMocks {}
     }
 }
 
-impl Drop for SetupIndyMocks {
```

Comment:
Why do you think removing `Drop`s in favor of resetting global state on init is preferable? I thing the global state should be clear after a test finishes, not when a new one starts. It should not be the responsibility of a test to clear the global state: the global state is clear by default when the first test starts, and it can be other than clear only by virtue of a test not clearing it after it has run, and so since there should be no side effects of running a test on global state, the test should clear the global state after it has finished running.